docs/source/api/rasterio-udfs.rst

```diff
@@ -234,40 +234,118 @@ Firstly we will create a spark DataFrame from a directory of raster files.
     +-----------------------------------------------------------+------------------------------+-----------+---------------------+-------+-------+-----------+----------------------+-------------+-------+---------------------------------------------------------------------------------------------------------------+
 
 
-Next we will define a function that will write a given raster file to disk.
+Next we will define a function that will write a given raster file to disk. A "gotcha" to keep in mind is that you do
+not want to have a file context manager open when you go to write out its context as the context manager will not yet
+have been flushed.
 
 .. code-block:: python
 
-    import numpy as np
-    import rasterio
-    from rasterio.io import MemoryFile
-    from io import BytesIO
-    from pyspark.sql.functions import udf
-    from pathlib import Path
+    @udf("string")
+    def write_raster(raster, driver, file_id, fuse_dir):
+        from io import BytesIO
+        from pathlib import Path
+        from pyspark.sql.functions import udf
+        from rasterio.io import MemoryFile
+        import numpy as np
+        import rasterio
+        import shutil
+        import tempfile
+
+        # - [1] populate the initial profile
+        # # profile is needed in order to georeference the image
+        profile = None
+        with MemoryFile(BytesIO(raster)) as memfile:
+            with memfile.open() as dataset:
+              profile = dataset.profile
+
+        # - [2] get the correct extension
+        extensions_map = rasterio.drivers.raster_driver_extensions()
+        driver_map = {v: k for k, v in extensions_map.items()}
+        extension = driver_map[driver] #e.g. GTiff
+        file_name = f"{file_id}.{extension}"
+
+        # - [3] write local raster
+        # - this is showing a single band [1]
+        #   being written
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_path = f"{tmp_dir}/{file_name}"
+            # here you can update profile using .update method
+            # example https://rasterio.readthedocs.io/en/latest/topics/writing.html
+            with rasterio.open(
+                tmp_path,
+                "w",
+                **profile
+            ) as dst:
+                dst.write(raster,1) # <- adjust as needed
```

Comment:
we need to add 
`dst.crs=rasterio.crs.CRS.from_epsg(crs_id)`
I had this issue recently, profile doesnt contain CRS info.

Reply:
@mjohns-databricks

Reply:
Ok, let me work that in.